server/src/main/scala/org/http4s/server/middleware/DefaultHead.scala

```diff
@@ -14,19 +14,18 @@ import cats.implicits._
   * requiring more optimization should implement their own HEAD handler.
   */
 object DefaultHead {
-  def apply[F[_]: Monad](@deprecatedName('service) routes: HttpRoutes[F]): HttpRoutes[F] =
+  def apply[F[_], G[_], B](@deprecatedName('service) http: Kleisli[F, Request[G], B])(
+    implicit F: MonoidK[F],
+    G: Functor[G]): Kleisli[F, Request[G], B] = {
     Kleisli { req =>
-      req.method match {
-        case Method.HEAD =>
-          (routes <+> headAsTruncatedGet(routes))(req)
-        case _ =>
-          routes(req)
+      http(req) <+> {
+        req.method match {
+          case Method.HEAD => {
+            http(req.withMethod(Method.GET).withEmptyBody)
```

Comment:
This is going to invoke the service with a request with an empty body.  We want to send a `GET` request, but return a `response` with an empty body.  Furthermore, we want to drain the response body, so any resources evaluated with the route are released.  So I think we want to keep the `headAsTruncatedGet`.

Reply:
Ah, ok. Makes sense. Fixing.